agent/health_endpoint.go

```diff
@@ -198,8 +198,13 @@ func (s *HTTPServer) healthServiceNodes(resp http.ResponseWriter, req *http.Requ
 	var out structs.IndexedCheckServiceNodes
 	defer setMeta(resp, &out.QueryMeta)
 
-	if args.QueryOptions.UseCache {
-		raw, m, err := s.agent.cache.Get(cachetype.HealthServicesName, &args)
+	// Use the streaming cachetype if enabled.
+	if args.QueryOptions.UseCache || (s.agent.config.EnableBackendStreaming && args.MinQueryIndex > 0) {
+		cType := cachetype.HealthServicesName
+		if s.agent.config.EnableBackendStreaming {
```

Comment:
Should this also include the `&& args.MinQueryIndex > 0` additional check?

Reply:
I think not since it's in the condition above - if we ever get here (say if the condition above changes one day) and streaming is enabled we should certainly use the streaming cache type.